### PR TITLE
Add initial news reader example

### PR DESCRIPTION
**Pull request by pcmoritz**
This is the initial implementation of the news reader example. It currently allows adding and reading RSS feeds, liking articles and storing article urls and likes in a sqlite3 database.

### CODE REVIEW

File: examples/newsreader/server.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+# instantiate the app
+app = Flask(__name__)
+app.config.from_object(__name__)
+
+# enable CORS
+CORS(app)
+
+
+@app.route('/api', methods=['POST'])
```
Comment (robertnishihara) [inline examples/newsreader/server.py]: `'` -> `"` everywhere in this file

File: examples/newsreader/server.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+def dispatcher():
+    req = request.get_json()
+    method_name = req["method_name"]
+    method_args = req["method_args"]
+    if hasattr(dispatcher.server, method_name):
+        method = getattr(dispatcher.server, method_name)
+        # Doing a blocking ray.get right after submitting the task
+        # might be bad for performance if the task is expensive.
+        result = ray.get(method.remote(*method_args))
+        return jsonify(result)
```
Comment (robertnishihara) [inline examples/newsreader/server.py]: Is this different from `json.dumps`?

File: examples/newsreader/server.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import sqlite3
+
+import ray
+
+
+@ray.remote
+class NewsServer(object):
+
+    def __init__(self):
+        self.conn = sqlite3.connect('newsreader.db')
```
Comment (robertnishihara) [inline examples/newsreader/server.py]: `'` -> `"` everywhere in this file

File: doc/source/example-newsreader.rst [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+.. code-block:: bash
+
+  pip install atoma
+  pip install flask
+
+
+To use this example you need to
+
+* Start the server with ``python server.py``
```
Comment (robertnishihara) [inline doc/source/example-newsreader.rst]: Make it clear which directory the user needs to be in to run this.

File: doc/source/example-newsreader.rst [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+* Clone the client code with ``git clone https://github.com/ray-project/qreader``
+* Start the client with ``cd qreader;  npm install; npm run dev``
+* You can now add a channel by clicking "Add channel" and for example pasting
+  ``http://news.ycombinator.com/rss`` into the field.
+* Star some of the articles and dump the database with ``sqlite3 newsreader.db``
```
Comment (robertnishihara) [inline doc/source/example-newsreader.rst]: Make it clear that this should be run from a terminal. I tried pasting this into the search bar in the web app.

File: examples/newsreader/server.py (hunk 2 above)
Comment (pcmoritz) [reply examples/newsreader/server.py]: https://stackoverflow.com/questions/7907596/json-dumps-vs-flask-jsonify

File: examples/newsreader/server.py (hunk 1 above)
Comment (pcmoritz) [reply examples/newsreader/server.py]: done

File: examples/newsreader/server.py (hunk 3 above)
Comment (pcmoritz) [reply examples/newsreader/server.py]: done

File: doc/source/example-newsreader.rst (hunk 4 above)
Comment (pcmoritz) [reply doc/source/example-newsreader.rst]: done

File: doc/source/example-newsreader.rst (hunk 5 above)
Comment (pcmoritz) [reply doc/source/example-newsreader.rst]: done

**Review by robertnishihara — APPROVED** [review]

**Comment by sylvoslee** [comment]
What's the point of doing this?

**Pull request merged**